app/src/modules/settings/routes/theme/theme-editor.vue

```diff
@@ -0,0 +1,277 @@
+		const options = field.meta?.options;
+		if (!options || options === {}) continue;
+		if (!dependencies[field.field]) {
+			dependencies[field.field] = [];
+		}
+		if (options.source) {
+			dependencies[field.field].push(options.source);
+		}
+		if (options.backgroundSource) {
+			dependencies[field.field].push(options.backgroundSource);
+		}
+	}
+	return dependencies;
+});
+
+const pendingChanges = ref<Record<string, any> | null>(null);
+
+const hasEdits = computed(() => pendingChanges.value !== null && Object.keys(pendingChanges.value).length > 0);
+
+const saving = ref(false);
+
+useShortcut('meta+s', () => {
+	if (hasEdits.value) save();
+});
+
+const { confirmLeave, leaveTo } = useEditsGuard(hasEdits);
+
+onBeforeRouteLeave(() => {
+	themeStore.setEditingTheme('light');
+	themeStore.setAppTheme();
+	pendingChanges.value = {};
+});
+
+watch(
+	() => route.params.theme as string,
+	(newTheme) => {
+		if (!newTheme) return;
+		themeStore.setEditingTheme(newTheme);
+		themeStore.setAppTheme(newTheme);
+		themeStore.populateFonts(newTheme);
+		themeFields.value = themeStore.getFields;
+		initialValues.value = themeStore.getInitialValues;
+		pendingChanges.value = {};
+	}
+);
+
+function updateChanges(newEdits: Record<string, any>) {
+	const toRegenerate = [];
+	for (const field of Object.keys(generatedFieldDependencies.value)) {
+		const dependencies = generatedFieldDependencies.value[field];
+		for (const dep of dependencies) {
+			if (
+				(newEdits?.[dep] && (!pendingChanges.value || newEdits?.[dep] !== pendingChanges.value[dep])) ||
+				(!newEdits?.[dep] && pendingChanges?.value?.[dep])
+			) {
+				toRegenerate.push(field);
+			}
+		}
+	}
+
+	if ((!pendingChanges.value || pendingChanges.value === {}) && toRegenerate.length > 0) {
+		pendingChanges.value = {};
+	}
+
+	if (!newEdits) {
+		newEdits = {};
+	}
+
+	for (const field of toRegenerate) {
+		const fieldData = themeFields.value.find((tField) => {
+			return tField.field === field;
+		});
+
+		if (!fieldData) continue;
+
+		const type = fieldData.meta?.options?.generateType;
+		const sourcePath = fieldData.meta?.options?.source || '';
+		const backgroundPath = fieldData.meta?.options?.backgroundSource || '';
+
+		if (!sourcePath) continue;
+
+		const sourceValue = newEdits?.[sourcePath] || initialValues.value[sourcePath];
+		const backgroundValue = newEdits?.[backgroundPath] || initialValues.value[backgroundPath];
+
+		const generatedColor = generateColor(type, sourceValue, backgroundValue || null);
+
+		if (initialValues.value[field].toLowerCase() === generatedColor.toLowerCase()) {
+			delete newEdits[field];
+		} else {
+			newEdits[field] = generatedColor;
+		}
+	}
+
+	pendingChanges.value = newEdits as Record<string, any>;
+}
+
+function generateColor(type: string, source: string, background?: string) {
+	let newColor = '#cccccc';
+	if (type === 'accent') {
+		newColor = generateAccent(source, editingTheme.value === 'dark' ? false : true);
```

Comment:
The component is trying to generate an accent color on each keystroke in the "Edit Raw Value" popup resulting in uncaught console errors. I believe validation if `source` is a valid hex code is in order here.
This also breaks the app when trying to manually add a non-hex color like `rgba(0,0,0,0)` having to refresh the page.

Reply:
Good catch, I didn't realize `Edit Raw Value` emitted updates on each keystroke. You're correct that a simple validation of `source` should fix it.

Reply:
Tried implementing this change and it didn't help. After a bit more research, I found that the existing color interface actually has a similar issue. Editing the raw value throws errors when it's not a valid color. I will investigate further.